neurons/validators/src/core/validator.py

```diff
@@ -298,7 +324,7 @@ async def sync(self):
             # fetch miners
             miners = self.fetch_miners(subtensor)
 
-            if await self.should_set_weights(subtensor):
+            if await self.should_set_weights(subtensor) or not self.set_weights_success:
```

Comment:
@LegendWorrier why we are checking the success status False here?

Reply:
![image](https://github.com/user-attachments/assets/81351bd6-a8db-4204-814d-dbbdbfa2afcb)
so, I did try include retry when setting weights is failed.
for that, I define self.set_weights_success  member variable.
if setting weight in fail, self.set_weights_success will set as false or if not,  true.

Reply:
@LegendWorrier so if the success flag is True, then you are not setting weights?

Reply:
if self.should_set_weights(subtensor) is true, it will set the weights.